### DISCUSSION TITLE
problem downloading files

### DISCUSSION THREAD

**Discussion by Graciasjr** [pocketbase/pocketbase#2761] [Q&A] [answered]
hi community.
 I'm trying to download a file whose resource is available on Pocketbase. I believe I have respected all the API rules and I even specified the "download" attribute in my `<a>` tag. Despite this, it redirects me to the resource to view it instead of downloading it. I would like to know if it's a problem at Pocketbase that I have potentially omitted or it's just me rambling 😭😹. 

 Script : 
```
<script>
Let url = `http://127.0.0.1:3001/api/files/${collectionName}/${id}/${filename}`;
</script>

<a href={url} download>Download<a/>
```

**Comment by ganigeorgiev** [comment, marked as answer]
Previaw-able resources (eg. images) are served with `Content-Disposition: inline` header.

The `download` attribute should work if it is from the same origin (in your case that doesn't seem the case because PocketBase will serve files from http://127.0.0.1:3001 and your Svelte app is probably on some other port). See also the notes in the [MDN `download` docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#download).

There are several workarounds if you want to always enforce a file download no matter of where the 2 applications reside, you can:

- download the file with JavaScript eg. using `fetch` and create a Blob (_for larger files, eg. 10mb+, this may cause some performance issues in the client browser_)
- create a hidden dummy iframe with `src` set to the file url
- if you use PocketBase as framework, attach to the [`OnFileDownloadRequest`](https://pocketbase.io/docs/event-hooks/#file-api-hooks) and overwrite the `Content-Disposition` header value to `attachment`

**Reply by ganigeorgiev** [reply]
_I'll consider for the next release to add an optional query parameter `download` to enforce the `attachment` value._

**Reply by Graciasjr** [reply]
Oh thank you very much I did not know that there were so many specificities around the downloading of files. You save my life THANKS.
